### PR TITLE
Added support for 32-bits samples in WAV files

### LINKED ISSUE #834: WAV files with 32 bits per file don't work with sf::Music when 16 bits per file WAV files do.
Here are two example files:

 * [32 bits per file example][1]
 * [16 bits per file example][2]

The easiest way to test these is by replacing the orchestral.ogg file with these and then run the sound example.

The two sample files were generated using libsndfile based on an original WAV file that I have. If you have any more questions then please let me know.

 [1]: https://s3.amazonaws.com/uploads.hipchat.com/10804/83693/0faMzNh4zU8d09A/out-1.wav
 [2]: https://s3.amazonaws.com/uploads.hipchat.com/10804/83693/LCNBWx7dNB7BreK/out-1.wav

### PR DESCRIPTION
**Pull request by LaurentGomila**
Fixes #834.

### CODE REVIEW

File: src/SFML/Audio/SoundFileReaderWav.cpp [hunk 1]
```diff
@@ -65,6 +65,17 @@ namespace
         return true;
     }
 
+    bool decode(sf::InputStream& stream, sf::Int32& value)
+    {
+        unsigned char bytes[sizeof(value)];
+        if (stream.read(bytes, sizeof(bytes)) != sizeof(bytes))
```
Comment (Bromeon) [inline src/SFML/Audio/SoundFileReaderWav.cpp]: Are the whole `sizeof`s not a bit confusing?

`sizeof(sf::Int32)` is 4 by definition, and in line 74, you assume 4 bytes anyway. I think the code might be more readable with a hardcoded `4`. Now it feels like the actual number of bytes could change, which is not true -- if it were, then we should seriously reconsider the definition of `sf::Int32` type.

Comment (LaurentGomila) [inline src/SFML/Audio/SoundFileReaderWav.cpp]: > Now that I had a look at the rest of the code, I see why you did it like this: you could copy-paste the other implementations

Yep :stuck_out_tongue:

**Comment by binary1248** [comment]
:+1:

**Comment by robertmassaioli** [comment]
Tested this branch on my files. It works! Thanks. :)

**Comment by eXpl0it3r** [comment]
This PR has been added to my merge list, meaning it will be merged soon, unless someone raises any concerns.

**Comment by robertmassaioli** [comment]
Yay, merged! Thank you team!

**Pull request merged**